### DISCUSSION TITLE
"not solvable" Error in Helmert Transformation

### DISCUSSION THREAD

**Discussion by ParisaShafiei** [groops-devs/groops#227] [Q&A]
Dear developers, 

I have a question about the Helmert transformation. I remember that in discussion [128](https://github.com/groops-devs/groops/discussions/128), @zhedumi explained the importance of estimating the Helmert transformation parameters in orbits comparison and I applied that for my network processing in comparing the GNSS satellite orbits. 

Now, I have two orbits that I'd like to compare and check whether there whether there is a global systematic component between the two orbits or not. So, for one example day, I have:

1) GRACE -C kinematic orbit processed at TUG
and
2) GRACE - C kinematic orbit processed by me

So, I used ["InstrumentEstimateHelmertTransformation"](https://groops-devs.github.io/groops/html/InstrumentEstimateHelmertTransformation.html) to get both the orbit differences and the Helmert parameters (please look at the screenshot).

![Screenshot from 2023-11-26 14-26-53](https://github.com/groops-devs/groops/assets/99748448/ad756173-d7db-4150-8359-c08ec1ed21ff)

By doing this, I got this error in the terminal:
![Screenshot from 2023-11-26 14-27-57](https://github.com/groops-devs/groops/assets/99748448/596752c6-0e26-476c-a363-4feea00b495e)

I already used this program for comparing GNSS orbits which in that case a loop over the PRN numbers was set to read the data. Here, I have not set any loops, because I only want to see the Helmert transformation parameters between two daily files, but I did not succeeded to do so.

My question is how should I set up this program to compare two daily orbits? and how do you use it in comparing the LEO orbits processed at different centers?

Thank you in advance for you support and I am looking forward to hearing from you.

With kind regards, 
Parisa

**Comment by zhedumi** [comment]
HI Parisa,

My advice is more in regards to global GNSS network estimation since everyday solution is basically its own GNSS reference frame solution that are aligned but not the same basically. Otherwise in the daily solutions you will have global variations in orbits and stations which you remove by helmert transformation. So the helmert transformation just makes sure that your solutions are in the same frame as the other solution.

But in this case I presume the same stations/satellites/models were used so I would just straight up compare them with the attitudes in my oppinion.
 Otherwise with just 2 LEO satellites you cannot estimate the 7 parameters from the Helmert transformation since you have to few observations (X,Y,Z in solution A and solution B).


Best regards,

Patrick

**Comment by ParisaShafiei** [comment]
Hi Patrick, 

Thank you for your reply. 

Correct, that is the same understanding that I had for applying the Helmert transformation, but it was a bit vague for me whether I have to apply that in comparing the LEO orbits as well or not. 

Besides, I am wondering what if different models that were used in processing the dataset are different? For example, in part of my processing, I also compare the orbits processed by GROOPS with AIUB solutions of the LEO satellite, e.g. GRACEC. In this case, can I just get only the 3 translation parameters instead of 7 parameters?

I tried this and when I just set options as you can see in the screenshot, it gives me the translation parameters, but I was doubtful if this is correct or not!
![Screenshot from 2023-11-28 17-39-04](https://github.com/groops-devs/groops/assets/99748448/75487fdc-a033-47ca-a787-d69c2dcefd3f)

Thanks again for your time and I am looking forward for your reply :)

With kind regards, 
Parisa
